docs/api-spec.md

```diff
@@ -0,0 +1,324 @@
+| Watch (GET)               | OPTIONAL         |
+| DeleteCollection (DELETE) | OPTIONAL         |
+
+\* Kubernetes only allows JSON merge patch for CRD types. It is recommended that if allowed, at least JSON Merge patch be made available. [JSON Merge Patch Spec (RFC 7386)](https://tools.ietf.org/html/rfc7386)
+
+## Detailed Resources - v1beta1
+
+The following schema defines a set of REQUIRED or RECOMMENDED resource fields on the Tekton resource types. Whether a field is REQUIRED or RECOMMENDED is denoted in the "Requirement" column.
+
+Additional fields MAY be provided by particular implementations, however it is expected that most extension will be accomplished via the `metadata.labels` and `metadata.annotations` fields, as Tekton implementations MAY validate supplied resources against these fields and refuse resources which specify unknown fields.
+
+Tekton implementations MUST NOT require `spec` fields outside this implementation; to do so would break interoperability between such implementations and implementations which implement validation of field names.
+
+**NB:** All fields and resources not listed below are assumed to be **OPTIONAL**, not RECOMMENDED or REQUIRED. For example, at this time, support for `PipelineRun`s and for CRUD operations on `Task`s or `Pipeline`s is **OPTIONAL**.
+
+### `TaskRun`
+
+#### Metadata
+
+Standard Kubernetes [meta.v1/ObjectMeta](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#objectmeta-v1-meta) resource.
+
+| Field Name           | Field Type         | Requirement |
+|----------------------|--------------------|-------------|
+| `name`               | string             | REQUIRED    |
+| `labels`             | map<string,string> | REQUIRED    |
+| `annotations`        | map<string,string> | REQUIRED    |
+| `creationTimestamp`* | string             | REQUIRED    |
+| `uid`                | string             | RECOMMENDED |
+| `resourceVersion`    | string             | RECOMMENDED |
+| `generation`         | int64              | RECOMMENDED |
+| `namespace`          | string             | RECOMMENDED |
+| `generateName`**     | string             | RECOMMENDED |
+
+\* `creationTimestamp` MUST be populated by the implementation, in [RFC3339](https://tools.ietf.org/html/rfc3339).
+
+** If `generateName` is supported by the implementation, when it is specified at creation, it MUST be prepended to a random string and set as the `name`, and not set on the subsequent response.
+
+#### Spec
+
+| Field Name            | Field Type           | Requirement |
+|-----------------------|----------------------|-------------|
+| `params`              | `[]Param`            | REQUIRED    |
+| `taskSpec`            | `TaskSpec`           | REQUIRED    |
+| `workspaces`          | `[]WorkspaceBinding` | REQUIRED    |
+| `status`              | Enum:<br>- `""` (default)<br>- `"TaskRunCancelled"` | RECOMMENDED |
+| `timeout`             | string (duration)    | RECOMMENDED |
```

Comment:
I have mixed feelings about this. Timeouts are key to the usability of the platform... if you want to keep this optional we could mention that the platform should at least provide some timeout mechanism (which would not be workflow specific).

Reply:
I agree! I wanted to have the initial spec be as minimally-scoped as possible, and timeout was on the bubble.

I could add a note that platforms should consider enforcing timeouts in some way, but honestly that kinda seems like their business. If they don't want to support timeout, I'm not sure we need to make them? I'm on the fence. 🤷‍♂️